configuration/configuration.go

```diff
@@ -202,6 +202,9 @@ type Configuration struct {
 				Deny []string `yaml:"deny,omitempty"`
 			} `yaml:"urls,omitempty"`
 		} `yaml:"manifests,omitempty"`
+		// RepositoryClass validation checks that the class of the configurations inside the
+		// the manifest matches the authorized resource when provided.
+		RepositoryClass bool `yaml:"repositoryclass,omitempty"`
```

Comment:
> A list of allowed classes?

This seems like the natural model.

> We could create a policy section for options like these. However I don't see it as too much different from the URLs allow/deny, also policies.

allow/deny controls which content gets into the backend, based on validation, whereas this is an enforced policy mechanism. While it does operate on content, the nature of that content policy is left to the backend. Indeed, this is subtle and this new feature really operates on both.

Another way of putting is that one is auth dependent while the other is validation dependent.

I'm not entirely sure.